compiler/rustc_span/src/symbol.rs

```diff
@@ -1195,6 +1195,7 @@ symbols! {
         if_let_rescope,
         if_while_or_patterns,
         ignore,
+        immediate_abort,
```

Comment:
I wonder if the symbol shouldn't be `immediate-abort`, to follow the CLI.

Which would make the cfgs be like `#[cfg(panic = "immediate-abort")]` which seems nicer than `#[cfg(panic = "immediate_abort")]`.

```suggestion
        immediate_abort: "immediate-abort",
```

Reply:
I suspect that if this is ever stabilized we will need some kind of special handling for the underscore version; we have `-Cdebug-assertions` which sets `cfg(debug_assertions)` and many other such flags.